modules/clone.sh

```diff
@@ -1,49 +1,68 @@
 function clone {
-  local device a b
+  local argument a b
   checkrpi
   checkroot
   checkargn $# 1
-  device="$1"
-  if [ -z "$device" ]; then
-    device="/dev/sdb"
-  fi
-
-  a=$(fdisk -l |grep /dev/mmcblk0: | grep -P '\d+ (?=bytes)' -o)
-  #echo "$a - /dev/mmcblk0"
-
-  b=$(fdisk -l |grep "$device": | grep -P '\d+ (?=bytes)' -o)
-  #echo "$b - /dev/sdb"
-
-  if [ -z "$a" ] || [ -z "$b" ]; then
-    echo "Error: the device $device wasn't detected. Please use 'fdisk -l' to find the device name."
-    return 1
-  fi
-
-  if [ $b -lt $a ]; then
-    echo "Error: the device $device is not big enough"
-    return 1
-  fi
-
-  if [ $a -eq $b ] || [ $a -lt $b ]; then
-    echo "copying...."
-    echo u > /proc/sysrq-trigger
-    dd if=/dev/mmcblk0 bs=1M of="$device" status=progress
-  fi
-
-  echo ; echo "A reboot is needed to re-enable write permissions to OS."
+  argument="$1"
+  path=$(fdisk -l | grep -o '^/dev/sd[a-z]' | sort -u)
```

Comment:
The use of `fdisk -l` piped to the grep command will never output since the `^` will try to find a line beginning with `/dev/`; `fdisk -l` will begin the line with Disk.

Also `fdisk -l` requires sudo privileges (I think) which may lead to problems if not running as root.

Finally remember to surround the variable value with quotes. Without wrapped quotes, its going to word split.

Heres the piped commands I used for the hardrive path: `path="$(lsblk -lp | awk '{print $1}' | grep -E '/dev/sd.*')"`

Reply:
When I test these new piped commands, I get unnecessary duplicates of the same device:
`/dev/sdc /dev/sdc1 /dev/sdc2`

I'll see if I can get it to only display `/dev/sdc`

Reply:
I modified it to look like this:
`path="$(lsblk -lp | awk '{print $1}' | grep -o '/dev/sd[a-z]' | sort -u)"`

On testing it only echoed:
`/dev/sdc`